src/Output/Output_BasePowerSpectrum.cpp

```diff
@@ -65,49 +66,52 @@ void Output_BasePowerSpectrum( const char *FileName )
    const int NRecvSlice = MIN( List_z_start[MPI_Rank]+local_nz, NX0_TOT[2] ) - MIN( List_z_start[MPI_Rank], NX0_TOT[2] );
 
    double *PS_total     = NULL;
-   real   *RhoK         = new real [ total_local_size ];                         // array storing density
-   real   *SendBuf      = new real [ amr->NPatchComma[0][1]*CUBE(PS1) ];         // MPI send buffer for density
-   real   *RecvBuf      = new real [ NX0_TOT[0]*NX0_TOT[1]*NRecvSlice ];         // MPI recv buffer for density
+   real   *VarK         = new real [ total_local_size ];                         // array storing data
+   real   *SendBuf      = new real [ amr->NPatchComma[0][1]*CUBE(PS1) ];         // MPI send buffer for data
+   real   *RecvBuf      = new real [ NX0_TOT[0]*NX0_TOT[1]*NRecvSlice ];         // MPI recv buffer for data
    long   *SendBuf_SIdx = new long [ amr->NPatchComma[0][1]*PS1 ];               // MPI send buffer for 1D coordinate in slab
    long   *RecvBuf_SIdx = new long [ NX0_TOT[0]*NX0_TOT[1]*NRecvSlice/SQR(PS1) ];// MPI recv buffer for 1D coordinate in slab
 
    int  *List_PID    [MPI_NRank];   // PID of each patch slice sent to each rank
    int  *List_k      [MPI_NRank];   // local z coordinate of each patch slice sent to each rank
-   int   List_NSend  [MPI_NRank];   // size of data (density) sent to each rank
-   int   List_NRecv  [MPI_NRank];   // size of data (density) received from each rank
+   int   List_NSend  [MPI_NRank];   // size of data sent to each rank
+   int   List_NRecv  [MPI_NRank];   // size of data received from each rank
+   const bool InPlacePad  = true;   // padded the array for in-place real-to-complex FFT
 
    if ( MPI_Rank == 0 )    PS_total = new double [Nx_Padded];
 
 
 // 3. initialize the particle density array (rho_ext) and collect particles to the target level
-#  ifdef PARTICLE
-   const bool TimingSendPar_No = false;
-   const bool JustCountNPar_No = false;
-#  ifdef LOAD_BALANCE
-   const bool PredictPos       = amr->Par->PredictPos;
-   const bool SibBufPatch      = true;
-   const bool FaSibBufPatch    = true;
-#  else
-   const bool PredictPos       = false;
-   const bool SibBufPatch      = NULL_BOOL;
-   const bool FaSibBufPatch    = NULL_BOOL;
-#  endif
-
-   Prepare_PatchData_InitParticleDensityArray( 0 );
-
-   Par_CollectParticle2OneLevel( 0, _PAR_MASS|_PAR_POSX|_PAR_POSY|_PAR_POSZ|_PAR_TYPE, PredictPos, Time[0],
-                                 SibBufPatch, FaSibBufPatch, JustCountNPar_No, TimingSendPar_No );
-#  endif // #ifdef PARTICLE
+   if ( TVar == _TOTAL_DENS ) {
```

Comment:
This will cause a compilation problem when enabling `PARTICLE`. Change it to
```
if ( TVar == _TOTAL_DENS ) {
      Prepare_PatchData_InitParticleDensityArray( 0 );

      Par_CollectParticle2OneLevel( 0, _PAR_MASS|_PAR_POSX|_PAR_POSY|_PAR_POSZ|_PAR_TYPE, PredictPos, Time[0],
                                    SibBufPatch, FaSibBufPatch, JustCountNPar_No, TimingSendPar_No );
}
```

Reply:
Done.